sdk/python/kfp/cli/components.py

```diff
@@ -0,0 +1,352 @@
+# Copyright 2021 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import enum
+import pathlib
+import shutil
+import subprocess
+import tempfile
+from typing import Any, Optional
+
+_DOCKER_IS_PRESENT = True
+try:
+    import docker
+except ImportError:
+    _DOCKER_IS_PRESENT = False
+    pass
+
+import typer
+
+import kfp
+from kfp.v2.components import component_factory, utils
+
+_REQUIREMENTS_TXT = 'requirements.txt'
+
+_DOCKERFILE = 'Dockerfile'
+
+_DOCKERFILE_TEMPLATE = '''
+FROM {base_image}
+
+WORKDIR {component_root_dir}
+COPY requirements.txt requirements.txt
+RUN pip install --no-cache-dir -r requirements.txt
+{maybe_copy_kfp_package}
+RUN pip install --no-cache-dir {kfp_package_path}
+COPY . .
+'''
+
+_DOCKERIGNORE = '.dockerignore'
+
+_DOCKERIGNORE_TEMPLATE = '''
+component_metadata/
+'''
+
+
+class _Engine(str, enum.Enum):
+    """Supported container build engines."""
+    DOCKER = 'docker'
+    KANIKO = 'kaniko'
+    CLOUD_BUILD = 'cloudbuild'
+
+
+app = typer.Typer()
+
+
+def _info(message: Any):
+    info = typer.style('INFO', fg=typer.colors.GREEN)
+    typer.echo('{}: {}'.format(info, message))
+
+
+def _warning(message: Any):
+    info = typer.style('WARNING', fg=typer.colors.YELLOW)
+    typer.echo('{}: {}'.format(info, message))
+
+
+def _error(message: Any):
+    info = typer.style('ERROR', fg=typer.colors.RED)
+    typer.echo('{}: {}'.format(info, message))
+
+
+class _ComponentBuilder():
+    """Helper class for building containerized v2 KFP components."""
+
+    def __init__(self,
+                 component_module: pathlib.Path,
+                 kfp_package_path: Optional[pathlib.Path] = None):
+        """ComponentBuilder builds containerized components.
+
+        Args:
+            component_module: Path to a Python file containing one or more
```

Comment:
Can we also support a directory containing multiple files where some of them may contain one or more component?

Reply:
Done. PTAL.